### PR TITLE
String Matching Fix v2

### PR DESCRIPTION
**Pull request by bwrbur**

### CODE REVIEW

**Comment by Dimenpsyonal** [comment]
Buh.

**Comment by SuperCater** [comment]
😨

**Comment by WalkerOfBacon** [comment]
@bwrbur can you make sure it doesnt ruin other commands too

**Comment by ccuser44** [comment]
@bwrbur what does this exactly fix or do?

**Comment by bwrbur** [comment]
> @bwrbur what does this exactly fix or do?

#1326

**Comment by ccuser44** [comment]
@bwrbur  I think you're doing it totslly wrong

Rather than not matching the string you should sanitise player selector arguments instead

**Comment by WalkerOfBacon** [comment]
> I have tested commands such as `:volume 0.5` and they work.

This says otherwise
image

**Comment by WalkerOfBacon** [comment]
I used :m and ended it with a dot and I get this error

**Comment by WalkerOfBacon** [comment]
It only works if it doesnt end with a dot
image

File: MainModule/Server/Core/Functions.lua [hunk 1]
```diff
@@ -1316,7 +1316,7 @@ return function(Vargs, GetEnv)
 		end;
 
 		Trim = function(str)
-			return string.match(str, "^%s*(.-)%s*$")
+			return string.match(str, "^%s*([^%.]*%.?[^%.]*[^%.]+)%s*$")
```
Comment (watameln) [inline MainModule/Server/Core/Functions.lua:1319]: your ass does not know how thsese filters work

**Comment by kaiserandaxl** [comment]
Sir, it doesn't fucking work! Instead of deleting every "." how about this, check if it's the only fucking thing there, if it is then return nothing if putting just "." does it to everyone, use common sense idiot.

**Comment by Dimenpsyonal** [comment]
True

**Pull request closed without merging**